tests/test_repository.py

```
@@ -0,0 +1,209 @@
+        await g.get_directory()
+
+        assert mock.await_count == 1
+        assert "git clone prefect" in mock.await_args[0][0]
+
+    async def test_reference_default(self, monkeypatch):
+        """Ensure that default command is 'git clone <repo name> -b <reference> --depth 1'  # noqa: E501
+        when just a repository and reference are given.
+        """
+
+        class p:
+            returncode = 0
+
+        mock = AsyncMock(return_value=p())
+        monkeypatch.setattr(prefect_github.repository, "run_process", mock)
+        g = GitHubRepository(repository="prefect", reference="2.0.0")
+        await g.get_directory()
+
+        assert mock.await_count == 1
+        assert "git clone prefect -b 2.0.0 --depth 1" in mock.await_args[0][0]
+
+    async def test_token_added_correctly_from_credential(self, monkeypatch):
+        """Ensure that the repo url is in the format `https://<oauth-key>@github.com/<username>/<repo>.git`."""  # noqa: E501
+
+        class p:
+            returncode = 0
+
+        mock = AsyncMock(return_value=p())
+        monkeypatch.setattr(prefect_github.repository, "run_process", mock)
+        credential = GitHubCredentials(token="XYZ")
+        g = GitHubRepository(
+            repository="https://github.com/PrefectHQ/prefect.git",
+            credentials=credential,
+        )
+        await g.get_directory()
+        assert mock.await_count == 1
+        assert (
+            "git clone https://XYZ@github.com/PrefectHQ/prefect.git"
+            in mock.await_args[0][0]
+        )
+
+    async def test_ssh_fails_with_credential(self, monkeypatch):
+        """Ensure that credentials cannot be passed in if the URL is not in the HTTPS
+        format.
+        """
+
+        class p:
+            returncode = 0
+
+        mock = AsyncMock(return_value=p())
+        monkeypatch.setattr(prefect_github.repository, "run_process", mock)
+        credential = GitHubCredentials(token="XYZ")
+        with pytest.raises(ValueError):
```

Maybe this can raise `InvalidRepositoryURLError` instead with a `match=`

---

The pydantic validation process eats the actual error, but I've updated it to be a `ValidationError` and a check for the string, which includes `(type=value_error.invalidrepositoryurl)`